collectors/ebpf.plugin/ebpf_mount.c

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "ebpf.h"
+#include "ebpf_mount.h"
+
+static ebpf_local_maps_t mount_maps[] = {{.name = "tbl_mount", .internal_input = NETDATA_MOUNT_END,
+                                          .user_input = 0, .type = NETDATA_EBPF_MAP_STATIC,
+                                          .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED},
+                                         {.name = NULL, .internal_input = 0, .user_input = 0,
+                                          .type = NETDATA_EBPF_MAP_CONTROLLER,
+                                          .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED}};
+
+static ebpf_data_t mount_data;
+static char *mount_dimension_name[NETDATA_EBPF_MOUNT_SYSCALL] = { "mount", "umount" };
+static netdata_syscall_stat_t mount_aggregated_data[NETDATA_EBPF_MOUNT_SYSCALL];
+static netdata_publish_syscall_t mount_publish_aggregated[NETDATA_EBPF_MOUNT_SYSCALL];
+
+struct config mount_config = { .first_section = NULL, .last_section = NULL, .mutex = NETDATA_MUTEX_INITIALIZER,
+                               .index = {.avl_tree = { .root = NULL, .compar = appconfig_section_compare },
+                                         .rwlock = AVL_LOCK_INITIALIZER } };
+
+static int read_thread_closed = 1;
+static netdata_idx_t *mount_values = NULL;
+
+static struct bpf_link **probe_links = NULL;
+static struct bpf_object *objects = NULL;
+
+static netdata_idx_t mount_hash_values[NETDATA_MOUNT_END];
+
+struct netdata_static_thread mount_threads = {"MOUNT KERNEL",
+                                              NULL, NULL, 1, NULL,
+                                              NULL,  NULL};
+
+/*****************************************************************
+ *
+ *  FUNCTIONS TO CLOSE THE THREAD
+ *
+ *****************************************************************/
+
+/**
+ * Clean up the main thread.
+ *
+ * @param ptr thread data.
+ */
+static void ebpf_mount_cleanup(void *ptr)
+{
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+    if (!em->enabled)
+        return;
+
+    freez(mount_threads.thread);
+    freez(mount_values);
```

Comment:
These two can be NULL frees if the function is executed before `mount_collector` is called in the mount init thread, which can happen in some error cases.

Reply:
This is the motive we set a test to verify whether the collector was enabled before. To call `freez` with NULL value is not a problem, observe that here we are not calling `free(3)`.